File: frontend/src/types/childReport.ts
import { HomeworkUpload } from './homework';

export type SubjectProgress = {
  subject: string;
  level: string;
  progress_percentage: number;
  current_topic?: string | null;
  strong_area?: string | null;
  needs_review?: string | null;
  recent_improvement?: string | null;
  completed_lessons: number;
  assessment_count: number;
  chat_count: number;
  message_count: number;
  last_activity_at?: string | null;
};

export type AssessmentSummary = {
  id?: number | string | null;
  subject: string;
  estimated_level: string;
  score_label?: string | null;
  strengths: string[];
  learning_gaps: string[];
  recommended_progression: string[];
  recommended_next_topics: string[];
  parent_summary?: string | null;
  created_at?: string | null;
};

export type TutorSessionSummary = {
  thread_id: string;
  subject: string;
  topic?: string | null;
  title?: string | null;
  message_count: number;
  time_spent: string;
  hints_used: number;
  practice_attempts: number;
  improvement_status: string;
  next_step: string;
  last_activity_at?: string | null;
};

export type ChildReport = {
  child_id: string;
  child_name: string;
  grade_level: string;
  report_period: string;
  subject_filter: string;
  current_learning_level: string;
  last_updated_at?: string | null;
  lessons_completed: number;
  questions_practiced: number;
  assessment_status: string;
  overall_summary: string;
  weekly_progress: string;
  time_spent_learning: string;
  brain_break_summary: string;
  subject_progress: SubjectProgress[];
  recent_assessments: AssessmentSummary[];
  recent_tutor_sessions: TutorSessionSummary[];
  homework_uploads: HomeworkUpload[];
  strengths: string[];
  weak_areas: string[];
  recommended_next_steps: string[];
};

export type WeeklyReportEmailPreview = {
  child_id: string;
  child_name: string;
  parent_id: string;
  report_period: string;
  subject_line: string;
  greeting: string;
  summary: string;
  subject_progress: SubjectProgress[];
  strengths: string[];
  weak_areas: string[];
  recommended_next_steps: string[];
  brain_break_summary: string;
  generated_at: string;
  email_connected: boolean;
  email_note: string;
};

File: backend/app/schemas/child_report.py
from pydantic import BaseModel

from .homework import HomeworkHistoryItem


class SubjectProgress(BaseModel):
    subject: str
    level: str
    progress_percentage: int = 0
    current_topic: str | None = None
    strong_area: str | None = None
    needs_review: str | None = None
    recent_improvement: str | None = None
    completed_lessons: int = 0
    assessment_count: int = 0
    chat_count: int = 0
    message_count: int = 0
    last_activity_at: str | None = None


class AssessmentSummary(BaseModel):
    id: int | str | None = None
    subject: str
    estimated_level: str
    score_label: str | None = None
    strengths: list[str] = []
    learning_gaps: list[str] = []
    recommended_progression: list[str] = []
    recommended_next_topics: list[str] = []
    parent_summary: str | None = None
    created_at: str | None = None


class TutorSessionSummary(BaseModel):
    thread_id: str
    subject: str
    topic: str | None = None
    title: str | None = None
    message_count: int = 0
    time_spent: str = 'Not tracked yet'
    hints_used: int = 0
    practice_attempts: int = 0
    improvement_status: str = 'Progress details will appear as more sessions are saved.'
    next_step: str = 'Continue with one short practice session.'
    last_activity_at: str | None = None


class ChildReportResponse(BaseModel):
    child_id: str
    child_name: str
    grade_level: str
    report_period: str
    subject_filter: str
    current_learning_level: str
    last_updated_at: str | None = None
    lessons_completed: int = 0
    questions_practiced: int = 0
    assessment_status: str
    overall_summary: str
    weekly_progress: str
    time_spent_learning: str
    brain_break_summary: str
    subject_progress: list[SubjectProgress]
    recent_assessments: list[AssessmentSummary]
    recent_tutor_sessions: list[TutorSessionSummary]
    homework_uploads: list[HomeworkHistoryItem] = []
    strengths: list[str]
    weak_areas: list[str]
    recommended_next_steps: list[str]


class WeeklyReportEmailPreview(BaseModel):
    child_id: str
    child_name: str
    parent_id: str
    report_period: str = 'week'
    subject_line: str
    greeting: str
    summary: str
    subject_progress: list[SubjectProgress]
    strengths: list[str]
    weak_areas: list[str]
    recommended_next_steps: list[str]
    brain_break_summary: str
    generated_at: str
    email_connected: bool = False
    email_note: str = 'Weekly email sending is prepared, but no email provider is connected yet.'
